ExchangeUnion/xud-docker
# Add Tor support

This PR enables Tor by default for xud. The branch is ready for testing, but **the code is not ready for review, yet** - pending cleanup + addding support for Tor for lnd-btc and lnd-ltc.

### Steps to test
Make sure all the existing containers are destroyed:
`cd ~/.xud-docker/simnet && docker-compose down`

Clean all existing local configuration:
`cd ~ && sudo rm -Rf ~/.xud-docker`

Clone the repository
`git clone https://github.com/ExchangeUnion/xud-docker.git ~/xud-docker`

Init and fetch git submodules
`cd ~/xud-docker`

`git submodule init`

`git submodule update --recursive --remote`

Fetch the Tor feature branch:
`git fetch origin feat-tor-rebased-v2:feat-tor-rebased-v2`

Switch to Tor feature branch:
`git checkout feat-tor-rebased-v2`

Build local simnet images with Tor support:
`~/xud-docker/images/xud-simnet/build-local.sh`
`~/xud-docker/images/lnd/build-local.sh`
`~/xud-docker/images/lnd-simnet/build-local.sh`

Start a clean Docker environment pointing to Tor feature branch:
`bash ~/xud.sh -b feat-tor-rebased-v2`

Take a look at the logs:
`logs -f xud`
